packages/mdc-snackbar/mdc-snackbar.scss

```diff
@@ -21,152 +21,183 @@
-  }
+  // Prevent snackbar from A) receiving mouse events while closed, and B) being visible during elastic scrolling.
+  // We can't use `display: none` on the root or surface elements because it prevents CSS transitions from animating.
+  transform: translateY(200vh);
 
-  @media (min-width: $mdc-snackbar-tablet-breakpoint) {
-    min-width: 288px;
-    max-width: 568px;
-    border-radius: 2px;
-  }
-}
-
-.mdc-snackbar--align-start {
-  @media (min-width: $mdc-snackbar-tablet-breakpoint) {
-    @include mdc-rtl-reflexive-position(left, 24px);
-
-    bottom: 24px;
-    transform: translate(0, 200%);
-  }
+  // Ignore mouse events on the root layout element.
+  pointer-events: none;
 
-  @media (max-width: ($mdc-snackbar-tablet-breakpoint - 1)) {
-    bottom: 0;
-    left: 0;
-    width: 100%;
-    transform: translate(0, 100%);
-  }
+  // Mobile Safari only supports `rgba` values for this property; named values like `transparent` are ignored.
+  // From Apple's docs:
+  // > This property obeys the alpha value, if specified.
+  // > If you don’t specify an alpha value, Safari on iOS applies a default alpha value to the color.
+  // > To disable tap highlighting, set the alpha value to 0 (invisible).
+  // > If you set the alpha value to 1.0 (opaque), the element is not visible when tapped.
+  // See https://github.com/ben-eb/postcss-colormin/issues/1
+  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
 }
 
-.mdc-snackbar--active {
-  transform: translate(0);
-  transition: mdc-animation-enter(transform, .25s);
-  pointer-events: auto;
-
-  &:not(.mdc-snackbar--align-start) {
-    transform: translate(-50%, 0);
-
-    @media (max-width: ($mdc-snackbar-tablet-breakpoint - 1)) {
-      bottom: 0;
-      left: 0;
-      width: 100%;
-      transform: translate(0);
-    }
-  }
+@at-root {
+  @include mdc-snackbar-fill-color(
+    $mdc-snackbar-overlay-fill-color,
+    $mdc-snackbar-overlay-fill-opacity,
+    $mdc-snackbar-surface-fill-color
+  );
+  @include mdc-snackbar-label-ink-color($mdc-snackbar-label-ink-color, $mdc-snackbar-label-ink-opacity);
+  @include mdc-snackbar-min-width($mdc-snackbar-min-width);
+  @include mdc-snackbar-max-width($mdc-snackbar-max-width);
+  @include mdc-snackbar-elevation($mdc-snackbar-elevation);
+  @include mdc-snackbar-shape-radius($mdc-snackbar-shape-radius);
 }
 
-.mdc-snackbar__action-wrapper {
-  @include mdc-rtl-reflexive-property(padding, 24px, 0);
+.mdc-snackbar--wide {
+  @include mdc-snackbar-viewport-margin($mdc-snackbar-viewport-margin-wide);
 }
 
-.mdc-snackbar--action-on-bottom {
-  flex-direction: column;
+.mdc-snackbar--leading {
+  justify-content: flex-start;
 }
 
-.mdc-snackbar__text {
-  @include mdc-typography(body1);
-  @include mdc-rtl-reflexive-box(margin, right, auto, ".mdc-snackbar");
+.mdc-snackbar--open,
+.mdc-snackbar--closing {
+  transform: none;
+}
 
+.mdc-snackbar__surface {
   display: flex;
+
+  // Non-static positioning is required to make `overflow: hidden` work in Safari.
```

Comment:
Are we really running into this bug? Wouldn't we want relative positioning anyway because the overlay element is absolutely-positioned inside the surface element?